File: database-driver-api/src/main/java/de/lino/database/export/transcript/format/PageFormat.java
package de.lino.database.export.transcript.format;

/**
 * A standard ISO 216 page size an export can be rendered at, e.g. by
 * {@code ExportCoordinator.TranscriptPDFExporter}, {@code ExportCoordinator.TranscriptExcelExporter}
 * or {@code ExportCoordinator.TranscriptDocxExporter} - every {@link de.lino.database.export.transcript.TranscriptExporter}
 * implementation accepts a {@link PageLayout}, but only these three actually render
 * pages and so are the only ones a chosen format has any visible effect on. Paired with
 * a {@link PageOrientation} via {@link PageLayout}.
 */
public enum PageFormat {

    /**
     * 297 x 420 mm.
     */
    A3,

    /**
     * 210 x 297 mm.
     */
    A4,

    /**
     * 148 x 210 mm.
     */
    A5

}

File: python/database-driver-api/src/database_driver/api/utils/export/transcript/format/page_format.py
"""Mirror of ``de.lino.database.utils.export.transcript.format.PageFormat``."""

from __future__ import annotations

from enum import Enum, unique


@unique
class PageFormat(Enum):
    """A standard ISO 216 page size an export can be rendered at.

    Every ``TranscriptExporter`` implementation accepts a ``PageLayout``, but only the
    PDF, Excel and DOCX exporters actually render pages and so are the only ones a chosen
    format has any visible effect on. Paired with a ``PageOrientation`` via
    ``PageLayout``.
    """

    A3 = "A3"
    """297 x 420 mm."""

    A4 = "A4"
    """210 x 297 mm."""

    A5 = "A5"
    """148 x 210 mm."""
